chainer/link.py

```diff
@@ -97,8 +120,9 @@ def __init__(self, **params):
         self._cpu = True
         self.name = None
 
-        for name, shape in six.iteritems(params):
-            self.add_param(name, shape)
+        for name, value in six.iteritems(params):
+            shape, dtype = _ensure_shape_dtype(value)
```

Comment:
It was merged from the branch of https://github.com/pfnet/chainer/pull/1633, it is now merged into master so the diff will go away after updating this PR.